### PR TITLE
src: add Daid's notes about the sprite slots

### PR DESCRIPTION
**Pull request by kemenaran**
Add @daid notes about how the sprite slots work.

TODO: 
- [x] rename related variables.

### CODE REVIEW

File: src/constants/memory/hram.asm [hunk 1]
```diff
@@ -48,20 +48,28 @@ hGameValuesSection::
 hNeedsUpdatingBGTiles::
  ds 1 ; FF90
 
-; bool value for enemie tile update
-; 0 = no update of enemie tiles needed,
-; 1 = update needed
-hNeedsUpdatingEnnemiesTiles::
+; Request the spriteslot at wC197 to be updated with the spritesheet
```
Comment (kemenaran) [inline src/constants/memory/hram.asm:51]: @daid I'm still not sure about the content of `wC197`. Is it the offset of the tile to be copied during the current stage, or a spriteslot index?

Comment (daid) [reply src/constants/memory/hram.asm:51]: $C197 is the sprite slot index. So target vram address is `addr = ($8400 + $100 * [$C197])` and `[wLoadedEntitySpritesheets + [$C197]]` contains the number of which spritesheet to load.

Those numbers are composed of 3 bits which indicate the bank, and 5 bits indicate which 16 tile row to take from that bank. The upper 3 bits are the index into `NpcTilesBankTable` to find which bank to use.

File: src/code/entities/hard_hat_beetle.asm [hunk 2]
```diff
@@ -1,3 +1,7 @@
+; Note: this entity, unlike most others, use 3 spriteslots (out of 4 available).
```
Comment (daid) [inline src/code/entities/hard_hat_beetle.asm:1]: The hardhat beetle isn't the enemy that has this property, it actually fits in a 2 slots.
The giant buzz blob has the property of using 3 slots as well as the stone hinox.

Other enemies that use more slots are:
- D4 boss (fish, always forget the name) but that can load everything due to a warp
- Evil Eagle, once again, loads everything on warp
- Nightmare, loads 4 sets on warp in, but switches out sprites even during battle (didn't check but I assume it uses the interrupt based mechanism)
- Slime boss (D3) uses 2 sets for the boss, but also 1 for the small slimes, so has 3 as well, but this one has the small slimes "preloaded" so it only needs to load 2 when entering the boss room.

There are a bunch of NPCs that have more then 2 spritesheets required as well, but those are all indoor, and thus behind warps (sun fish and manu are two examples)
The sneaky one here is the walrus, which uses 2 spritesheets, but then another one for Marin singing, so 3. It uses the same trick as the slime boss, so the walrus sprites are loaded 1 room before the actual walrus. So Marin's singing sprite can be loaded when entering the room with the walrus.

File: src/constants/memory/hram.asm (hunk 1 above)
Comment (kemenaran) [reply src/constants/memory/hram.asm]: Ok, got it. I updated the naming and the comments, I think it should be correct now.

**Comment by kemenaran** [comment]
I think this is ready to be merged.

Hopefully I will also be able to reformat Daid's notes into a section in the [High-level game engine documentation](https://github.com/zladx/LADX-Disassembly/wiki/Game-engine-documentation) at some point.

**Pull request merged**